src/NodeCollector/NodeAnalyzer/ArrayCallableMethodMatcher.php

```diff
@@ -188,9 +189,9 @@ private function resolveClassConstFetchType(
 
     private function resolveCallerType(Expr $expr, Scope $scope, ?string $classMethodName): Type
     {
-        if ($expr instanceof ClassConstFetch) {
-            // static ::class reference?
-            $callerType = $this->resolveClassConstFetchType($expr, $scope, $classMethodName);
+        if ($expr instanceof ClassConstFetch || $expr instanceof Class_) {
+            // class context means self|static ::class or __CLASS__
+            $callerType = $this->resolveClassContextType($expr, $scope, $classMethodName);
```

Comment:
I wanted to avoid using the abstract term `'context'`.
But I couldn't think of a better way to express it...